### PR TITLE
Avoid assuming the wave size during transformations

### PR DESCRIPTION
**Pull request by giuseros**
This is trying to avoid the issue of "assuming" `waveSize==64` causing issue when dealing with `wmma`. 

The only place where we still assume the wave size is in `MfmaInsnsGroup.h`. 

This is because `MfmaInsnsGroup` lives in `MLIRRockOps` and `MLIRUtility` uses `MLIRRockOps`, so I would create a cyclic dependency. 

For the time being, I decided to not care, because for now the `mfma` instruction is specific to architectures for which `waveSize` is 64.

### CODE REVIEW

File: mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp [hunk 1]
```diff
@@ -479,25 +480,25 @@ struct BlockwiseGemmV2RewritePattern
       }
     };
 
-    auto ldsToRegisterCopyKdim =
```
Comment (manupak) [inline mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp]: What is the change here ?

File: mlir/include/mlir/Dialect/Rock/Tuning/GridwiseGemmParams.h [hunk 2]
```diff
@@ -272,7 +270,8 @@ class PopulateParamsXDL : public BasePopulateParams<InitParamsXDL> {
 
   LogicalResult isValidBlockwiseGemmXDLOPS(const InitParamsXDL &param,
                                            Type dataTypeA, Type dataTypeB,
-                                           StringRef arch, uint32_t blockSize);
+                                           StringRef arch, int64_t waveSize,
```
Comment (manupak) [inline mlir/include/mlir/Dialect/Rock/Tuning/GridwiseGemmParams.h]: nit : given we look up waveSize using arch, the expansion of func signature is to save that lookup ?

**Review by manupak — APPROVED** [review]
Looks good with a question and a nit.

File: mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp (hunk 1 above)
Comment (krzysz00) [reply mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp]: I think it's clang-format

**Review by krzysz00 — APPROVED** [review]
Re moving around the MfmaInstrGroup, we can probably factor the `Rock` utilities - maybe move the AmdArchDb and other stuff like that into something like `mlir/lib/Util/`.

File: mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp (hunk 1 above)
Comment (giuseros) [reply mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp]: Yep, exactly

File: mlir/include/mlir/Dialect/Rock/Tuning/GridwiseGemmParams.h (hunk 2 above)
Comment (giuseros) [reply mlir/include/mlir/Dialect/Rock/Tuning/GridwiseGemmParams.h]: Yeah, I agree, it's probably better to look up again

**Comment by giuseros** [comment]
> Re moving around the MfmaInstrGroup, we can probably factor the `Rock` utilities - maybe move the AmdArchDb and other stuff like that into something like `mlir/lib/Util/`.

Yeah like a place where no rock ops are used so we don't have any cycles

**Pull request merged**